extensions/timeseries/src/main/java/ai/djl/timeseries/dataset/M5Forecast.java

```diff
@@ -0,0 +1,243 @@
+            cardinality = new ArrayList<>();
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        protected Builder self() {
+            return this;
+        }
+
+        /**
+         * Set the repository containing the path.
+         *
+         * @param repository the repository containing the path
+         * @return this builder
+         */
+        public Builder setRepository(Repository repository) {
+            this.repository = repository;
+            return this;
+        }
+
+        /**
+         * Set the optional usage.
+         *
+         * @param usage the usage
+         * @return this builder
+         */
+        public Builder optUsage(Usage usage) {
+            this.usage = usage;
+            return this;
+        }
+
+        /**
+         * Add a feature to the features set of the filed name.
+         *
+         * @param name the name of the feature
+         * @param fieldName the field name
+         * @return this builder
+         */
+        public Builder addFeature(String name, FieldName fieldName) {
+            return addFeature(name, fieldName, false);
+        }
+
+        /**
+         * Add a feature to the features set of the filed name with onehot encoding.
+         *
+         * @param name the name of the feature
+         * @param fieldName the field name
+         * @param onehotEncode true if use onehot encoding
+         * @return this builder
+         */
+        public Builder addFeature(String name, FieldName fieldName, boolean onehotEncode) {
+            parseFeatures();
+            if (mf.categorical.contains(name)) {
+                Map<String, Integer> map = mf.featureToMap.get(name);
+                if (map == null) {
+                    return addFieldFeature(
+                            fieldName,
+                            new Feature(name, Featurizers.getStringFeaturizer(onehotEncode)));
+                }
+                cardinality.add(map.size());
+                return addFieldFeature(fieldName, new Feature(name, map, onehotEncode));
+            }
+            return addFieldFeature(fieldName, new Feature(name, true));
+        }
+
+        /**
+         * Returns the available features of this dataset.
+         *
+         * @return a list of feature names
+         */
+        public List<String> getAvailableFeatures() {
+            parseFeatures();
+            return mf.featureArray;
+        }
+
+        /**
+         * Build the new {@code M5Forecast}.
+         *
+         * @return the new {@code M5Forecast}
+         */
+        @Override
+        public M5Forecast build() {
+            validate();
+            return new M5Forecast(this);
+        }
+
+        private void parseFeatures() {
+            if (mf == null) {
+                try (InputStream is =
+                                Objects.requireNonNull(
+                                        M5Forecast.class.getResourceAsStream("m5forecast.json"));
```

Comment:
This file is pretty big, we should put this file on S3

Reply:
Thanks, do I need to corporate with my mentor to upload this?